controller/controller.go

```diff
@@ -0,0 +1,147 @@
+package controller
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"github.com/dgrijalva/jwt-go"
+	"iotdashboard/dbmanager"
+	"golang.org/x/crypto/bcrypt"
+	"time"
+)
```

Comment:
Just as a little tip, it doesn't look like you're using `goimports` (can tell by the way these imports are organized). Check it out, it's `gofmt` on steroids and will make your life easier.

Reply:
Left this for last but it should have been the first thing I did. Thanks for the trick